doc/notes/189_reporting_ant.adoc

```diff
@@ -0,0 +1,224 @@
+* throughput (events per second) over a time period ("area under the curve" line graph)
+* jobs succeeded versus failed versus alarmed (pie chart:  green, yellow, red)
+* list of alarmed job names with counters next to them counting the alarms sorted by incidence
+* list of failed job names with counters next to them counting the failures sorted by incidence
+* list of successful job names with counters next to them counting them sorted by incidence
+* list of retried job names with counters next to them counting them sorted by incidence
+* proportion of happy versus unhappy jobs (pie chart:  blue, purple)
+
+===== 3.1.1.1 Instantaneous Throughput
+
+The instantaneous throughput will display the throughput of the
+Protocol Verifier "right now".  It may be time-averaged for smoothing but
+the time window will be short.  This could be displayed in something like
+a speedometer dial.
+
+The units shall be events per second.
+
+===== 3.1.1.2 Throughput Over Time
+
+A graph of the throughput of the Protocol Verifier shall be displayed
+supplying a view of the throughput over the last (say) 24 hours.
+Optionally the display window could be adjustable.  The graph may be
+coloured in an "area under the curve" fashion.
+
+The units shall be events per second.
+
+===== 3.1.1.3 Alarmed versus Failed versus Successful Jobs
+
+A relative proportion of jobs that have ended in alarm conditions, failure
+and success shall be displayed.  A pie chart with the different conditions
+shown as different colours (e.g. red for alarm, yellow for failure and
+green for success) shall make the proportions visually clear.
+
+The units shall be total alarm/failure/success over a noted time period
+(e.g. "today" or "last 24 hours").
+
+===== 3.1.1.4 List of Alarmed Jobs
+
+A list of job types that experienced alarm condition shall be displayed.
+The names of the jobs shall be shown in a readable format.  A counter of
+the number of alarms (over some period of time) shall be shown.  The list
+shall be sorted in descending order starting with the job type receiving
+the greatest incidence of alarms.
+
+===== 3.1.1.5 List of Failed Jobs
+
+A list of job types that failed shall be displayed.  The display format
+shall follow the format described in <<3.1.1.4 List of Alarmed Jobs>>.
+
+===== 3.1.1.6 List of Successful Jobs
+
+A list of job types that succeeded shall be displayed.  The display format
+shall follow the format described in <<3.1.1.4 List of Alarmed Jobs>>.
+
+===== 3.1.1.7 List of Retried Jobs
+
+A list of job types that experienced retries shall be displayed.  The
+names of the jobs shall be shown in a readable format.  A counter of the
+number of retries (over some period of time) shall be shown.  The list
+shall be sorted in descending order starting with the job type receiving
+the greatest incidence of retries.
+
+===== 3.1.1.8 Proportion of Happy versus Unhappy Jobs
+
+A relative proportion of happy jobs and unhappy jobs shall be displayed.
```

Comment:
Would I be correct in thinking that the happiness/unhappiness of jobs is independent of whether they are successful, failed or alarmed? - note that a 'successful' job could have been an unhappy job.

Reply:
Right.  A job always starts happy and migrates to unhappy when an expected error event is received.  This is the statistic I want.  The client is expecting roughly 1% of jobs to be failing in expected ways.